### PR TITLE
curses: various fixes

### PR DESCRIPTION
**Pull request by hauntsaninja**
- fix arg names
- fix types based on default value
- mark positional-only args

### CODE REVIEW

File: stdlib/2and3/curses/__init__.pyi [hunk 1]
```diff
@@ -12,4 +12,4 @@ COLS: int
 COLORS: int
 COLOR_PAIRS: int
 
-def wrapper(func: Callable[..., _T], *arg: Any, **kwds: Any) -> _T: ...
+def wrapper(__func: Callable[..., _T], *arg: Any, **kwds: Any) -> _T: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/curses/__init__.pyi]: I don't think this is positional-only?

```
In [27]: curses.wrapper??
Signature: curses.wrapper(func, *args, **kwds)
Source:   
def wrapper(func, *args, **kwds):
    """Wrapper function that initializes curses and calls another function,
```
I'm on 3.7

File: stdlib/2and3/curses/textpad.pyi [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-from typing import Callable, Union
+from typing import Callable, Optional, Union
 
 def rectangle(win: _CursesWindow, uly: int, ulx: int, lry: int, lrx: int) -> None: ...
 
 class Textbox:
     stripspaces: bool
-    def __init__(self, w: _CursesWindow, insert_mode: bool = ...) -> None: ...
-    def edit(self, validate: Callable[[int], int]) -> str: ...
+    def __init__(self, win: _CursesWindow, insert_mode: bool = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/curses/textpad.pyi]: This one isn't either `    def __init__(self, win, insert_mode=False):`.

Comment (hauntsaninja) [reply stdlib/2and3/curses/textpad.pyi]: It isn't positional-only in the stub either :-)

Comment (JelleZijlstra) [reply stdlib/2and3/curses/textpad.pyi]: um, I guess I just saw one underscore and assumed there were underscores everywhere

File: stdlib/2and3/curses/__init__.pyi (hunk 1 above)
Comment (hauntsaninja) [reply stdlib/2and3/curses/__init__.pyi]: The text signature has it as positional-only, https://github.com/python/cpython/blob/3.8/Lib/curses/__init__.py#L113
(It's also positional-only in 3.9, for whatever that's worth)

Comment (JelleZijlstra) [reply stdlib/2and3/curses/__init__.pyi]: I see, looks like they deprecated passing it as a keyword in 3.8 and removed it in 3.9. I suppose strictly we should mark it as positional-only in 3.9+ only, but it should be fine to just have it unconditionally positional-only, at least as long as nobody actually complains.

**Pull request merged**